state/spacesdiscovery.go

```diff
@@ -77,22 +78,12 @@ func (st *State) SaveSubnetsFromProvider(subnets []network.SubnetInfo, spaceName
 		if ip.IsInterfaceLocalMulticast() || ip.IsLinkLocalMulticast() || ip.IsLinkLocalUnicast() {
 			continue
 		}
-		var firstZone string
-		if len(subnet.AvailabilityZones) > 0 {
-			firstZone = subnet.AvailabilityZones[0]
-		}
-		_, err = st.AddSubnet(SubnetInfo{
-			ProviderId:        subnet.ProviderId,
-			ProviderNetworkId: subnet.ProviderNetworkId,
-			CIDR:              subnet.CIDR,
-			SpaceName:         spaceName,
-			VLANTag:           subnet.VLANTag,
-			AvailabilityZone:  firstZone,
-		})
+
+		subnet.SpaceName = spaceName
+		_, err = st.AddSubnet(subnet)
```

Comment:
This is a nice realisation of the change - we no longer convert `network.SubnetInfo` -> `state.SubnetInfo` -> `state.subnetDoc`.